### ISSUE TITLE
CLI UX improvement

### ISSUE THREAD

**Issue by samalba** [aweris/gale#33]
The CLI is not intuitive currently, both `--workflow` and `--job` are mandatory, so they should not be CLI options, but arguments instead. Here is a quick proposal:

1. Turn workflow and job into arguments

```
gale <workflow> <job>
```

And add info and examples in the help.

2. If workflow or job are not provided, introspect the current repository

Example:

```
$ gale
Workflow and job must be provided, detected the following workflows:

- .github/workflows/deploy.yaml
--- build    (invoke with `gale .github/workflows/deploy.yaml build`)
--- test     (invoke with `gale .github/workflows/deploy.yaml test`)
--- deploy   (invoke with `gale .github/workflows/deploy.yaml deploy`)
```

This is just a suggestion, I think `1.` is a must-have, `2.` is a nice-to-have and could be implemented differently. I just think it would save time to introspect a repo.

cc @aweris @tiborvass

**Comment by aweris** [comment]
What about re-structure cli like:

- `gale` - prints `help` text
- `gale run <workflow> <job>` - runs given workflow and job. **
- `gale list <workflow>` if workflow provided, list workflow information like jobs, otherwise lists all workflows in the current repository

I think having explicit commands and proper help text in default run would create better UX for users

** run interface will change overtime parallel to adding missing features

**Comment by aweris** [comment]
the suggestion above merged with #34.
